Syntax & execution CSharpRepl exceptions shouldn't be translated

### Version

0.6.5

### What happened?

Exceptions from CSharpRepl are translated, and they shouldn't be.
This happens when you try running code, which wouldn't be compiled, for example missing semicolon, no such variable.

Exceptions from the code itself aren't translated, which is expected.

Example:
```csharp
> System.Threading.Thread.CurrentThread.CurrentCulture
pl-PL
> System.Threading.Thread.CurrentThread.CurrentUICulture
en-US
> throw new Exception();
┌────────────────────Exception─────────────────────┐
│ Exception of type 'System.Exception' was thrown. │
└──────────────────────────────────────────────────┘

> AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName
".NETCoreApp,Version=v7.0"
> show translation issue
┌───────────────────Exception───────────────────┐
│ (1,18): error CS1002: Oczekiwano średnika (;) │
└───────────────────────────────────────────────┘
```

When I have the same code in a .cs file, and try compiling it either with Visual Studio 2022, or via `dotnet build` / `dotnet run`, I'm getting this message in English:

```
C:\Dev\TestApp\TestApp\Program.cs(7,38): error CS1002: ; expected [C:\Dev\TestApp\TestApp\TestApp.csproj]
```

The proper native .cs code, when run, shows the same culture values - pl-PL / en-US.

Running Windows 11, English United States with Polish language pack added.
Windows Display Language is set to English.
Preferred languages are English (United States), Polish.
Country or region: United States
Regional format: Polish (Poland)
Even the keyboard language was set to English (United States), with Polish Programmer's layout

.cs files tested with both .NET 7.0.12 / 7.0.402 and 8.0.0-rc.1.23419.4 / 8.0.100-rc.1.23455.8

I haven't tested CSharpRepl with #241 / #296 fix.

--- a/CSharpRepl.Services/Configuration.cs
+++ b/CSharpRepl.Services/Configuration.cs
@@ -170,7 +170,7 @@ public sealed class Configuration
             submitPrompt,
             triggerOverloadList: new(new KeyPressPattern('('), new KeyPressPattern('['), new KeyPressPattern(','), new KeyPressPattern('<')));
 
-        Culture = string.IsNullOrWhiteSpace(cultureName) ? CultureInfo.CurrentCulture : CultureInfo.GetCultureInfo(cultureName, true);
+        Culture = string.IsNullOrWhiteSpace(cultureName) ? CultureInfo.CurrentUICulture : CultureInfo.GetCultureInfo(cultureName, true);
     }
 
     public CultureInfo Culture { get; }

--- a/CSharpRepl/Program.cs
+++ b/CSharpRepl/Program.cs
@@ -4,6 +4,7 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
 using System.IO;
 using System.Text;
 using System.Threading.Tasks;
@@ -35,6 +36,8 @@ internal static class Program
         if (!TryParseArguments(args, configFile, out var config))
             return ExitCodes.ErrorParseArguments;
 
+        SetDefaultCulture(config);
+
         if (config.OutputForEarlyExit.Text is not null)
         {
             console.WriteLine(config.OutputForEarlyExit);
@@ -46,14 +49,14 @@ internal static class Program
         var roslyn = new RoslynServices(console, config, logger);
 
         // we're getting piped input, just evaluate the input and exit.
-        if(Console.IsInputRedirected)
+        if (Console.IsInputRedirected)
         {
             var evaluator = new PipedInputEvaluator(console, roslyn);
             return config.StreamPipedInput
-                ? await evaluator.EvaluateStreamingPipeInputAsync()
-                : await evaluator.EvaluateCollectedPipeInputAsync();
+                ? await evaluator.EvaluateStreamingPipeInputAsync().ConfigureAwait(false)
+                : await evaluator.EvaluateCollectedPipeInputAsync().ConfigureAwait(false);
         }
-        else if(config.StreamPipedInput)
+        else if (config.StreamPipedInput)
         {
             console.WriteErrorLine("--streamPipedInput specified but no redirected input received. This configuration option should be used with redirected standard input.");
             return ExitCodes.ErrorParseArguments;
@@ -107,18 +110,19 @@ internal static class Program
         return appStorage;
     }
 
+    private static void SetDefaultCulture(Configuration config)
+    {
+        CultureInfo.DefaultThreadCurrentUICulture = config.Culture;
+        // theoretically we shouldn't need to do the following, but in practice we need it in order to
+        // get compiler errors emitted by CSharpScript in the right language (see https://github.com/waf/CSharpRepl/issues/312)
+        CultureInfo.DefaultThreadCurrentCulture = config.Culture;
+    }
+
     /// <summary>
     /// Initialize logging. It's off by default, unless the user passes the --trace flag.
     /// </summary>
-    private static ITraceLogger InitializeLogging(bool trace)
-    {
-        if (!trace)
-        {
-            return new NullLogger();
-        }
-
-        return TraceLogger.Create($"csharprepl-tracelog-{DateTime.UtcNow:yyyy-MM-dd}.txt");
-    }
+    private static ITraceLogger InitializeLogging(bool trace) =>
+        !trace ? new NullLogger() : TraceLogger.Create($"csharprepl-tracelog-{DateTime.UtcNow:yyyy-MM-dd}.txt");
 
     private static (Prompt? prompt, int exitCode) InitializePrompt(IConsoleEx console, string appStorage, RoslynServices roslyn, Configuration config)
     {

